cassandra/util.py

```diff
@@ -1193,3 +1193,48 @@ def _sanitize_identifiers(field_names):
                 names_out[index] = "%s_" % (names_out[index],)
             observed_names.add(names_out[index])
     return names_out
+
+
+class Duration(object):
+    """
+    Cassandra Duration Type
+    """
+
+    months = 0
+    days = 0
+    nanoseconds = 0
+
+    def __init__(self, months=0, days=0, nanoseconds=0):
+        self.months = months
+        self.days = days
+        self.nanoseconds = nanoseconds
+        self.validate()
+
+    def validate(self):
+        """
+        A Duration is valid if its values are all positive or all negative. It cannot has mixed signs.
+        """
+        if self._has_negative_values and self._has_positive_values:
+            raise ValueError('Duration values cannot have mixed signs.')
+
+    @property
+    def _has_negative_values(self):
+        return self.months < 0 or self.days < 0 or self.nanoseconds < 0
+
+    @property
+    def _has_positive_values(self):
+        return self.months > 0 or self.days > 0 or self.nanoseconds > 0
+
+    def __eq__(self, other):
+        return isinstance(other, self.__class__) and self.months == other.months and self.days == other.days and self.nanoseconds == other.nanoseconds
+
+    def __repr__(self):
+        return "Duration({0}, {1}, {2})".format(self.months, self.days, self.nanoseconds)
+
+    def __str__(self):
+        return '{0}{1}mo{2}d{3}ns'.format(
```

Comment:
Fixing the use of %. I just always .format() and forgot % is used in all the package code base.

Also fixed the float/decimal using %d. Only integers are accepted. This will do the same thing than the binary encoding, which cast  to integers.